File: csharp/src/Ice/Internal/Time.cs
// Copyright (c) ZeroC, Inc.

using System.Diagnostics;

namespace Ice.Internal;

public static class Time
{
    static Time() => _stopwatch.Start();

    public static long currentMonotonicTimeMillis() => _stopwatch.ElapsedMilliseconds;

    private static readonly Stopwatch _stopwatch = new();
}

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/Time.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/**
 * @hidden Public because it's used by IceLocatorDiscovery and the android tests.
 */
public final class Time {
    public static long currentMonotonicTimeMillis() {
        return System.nanoTime() / 1000000;
    }
}
